skyxplore-ui-test/src/test/java/com/mentoring/test/EquipmentTest.java

```diff
@@ -1,61 +1,127 @@
+        } else {
+            log.warn("DB disconnection skipped. There is no database connection to disconnect.");
+        }
+    }
+
+    private void connectToDB() {
+        if(!Config.isLocalEnvironmentUsed()) {
+            throw new EnvironmentNotSupportedException("Only local environment is supported for database connection. " +
+                    "Skipping tests.");
+        }
+        database = new Database();
+    }
+
     @Description("Create and set up an equipment.")
     @Severity(SeverityLevel.BLOCKER)
     @Feature(EQUIPMENT)
     @Test(groups = {REGRESSION})
-    public void createAndSetUpAnEquipment() {
-        String validCharacterName = UserUtils.generateRandomCharacterName();
+    public void createAndEquipExtender() throws ExtraFreeSlotsNumberNotFoundException {
+        String characterName = UserUtils.generateRandomCharacterName();
         User user = new User();
 
         FactoryPage factoryPage = new IndexPage(driver)
                 .login(user)
-                .createNewCharacter(validCharacterName)
-                .selectCharacter(validCharacterName)
+                .createNewCharacter(characterName)
+                .selectCharacter(characterName)
                 .openFactoryPage()
                 .selectConnectorExtender()
                 .clickCex01ProductionButton();
-
         assertThat(factoryPage.isExtenderItemInQueueVisible())
-                .as("Extender item in queue should be visible.")
+                .as("Extender item in Queue should be visible.")
                 .isTrue();
-        assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
-                .as("Quantity in queue is not correct")
+        softAssertion.assertThat(factoryPage.getQuantityInQueue().equals(DEFAULT_QUANTITY))
+                .as("Quantity in Queue should be correct.")
                 .isTrue();
-        assertThat(factoryPage.isQueueProcessStarted())
-                .as("Queue process is not started.")
+        softAssertion.assertThat(factoryPage.isQueueProcessBarVisible())
+                .as("Queue process bar should be visible.")
                 .isTrue();
 
-        //TODO: Subtask-3: Production time of CEX-01 is updated and item is finished
-        //TODO: Subtask-4: CEX-01 item is added to the ship
+        log.info("Updating DB: Set queue process end-time to finish the production.");
+        factoryId = DbSelects.getFactoryId(database, user.getEmail(), characterName);
+        DbUpdates.setProductionEndTimeByFactoryId(database, factoryId, "0");
 
-        //softAssertion.assertAll();
-    }
+        driver.navigate().refresh();
+        assertThat(factoryPage.isQueueProcessBarVisible())
+                .as("Queue process bar should be updated")
+                .isTrue();
+        assertThat(factoryPage.isQueueDisappeared())
+                .as("Queue should be disappeared")
+                .isTrue();
+
+        EquipmentPage equipPage = new FactoryPage(driver)
```

Comment:
You already have the FactoryPage instantiated above, no need for a new. Because of this, you can keep FactoryPage package private.

Reply:
I see. Fixed, thx.